source/s3_copy_object.c

```diff
@@ -92,10 +91,22 @@ struct aws_s3_meta_request *aws_s3_meta_request_copy_object_new(
     copy_object->synced_data.content_length = UNKNOWN_CONTENT_LENGTH;
     copy_object->synced_data.total_num_parts = UNKNOWN_NUM_PARTS;
     copy_object->threaded_update_data.next_part_number = 1;
+    if (options->copy_source_uri.len != 0) {
+        if (aws_uri_init_parse(&copy_object->source_uri, allocator, &options->copy_source_uri)) {
+            AWS_LOGF_ERROR(
+                AWS_LS_S3_META_REQUEST,
+                "Unable to parse the copy_source_uri provided in the request: " PRInSTR "",
+                AWS_BYTE_CURSOR_PRI(options->copy_source_uri));
+            goto on_error;
```

Comment:
the `copy_object->synced_data.part_list` was initialized above, it needs to be cleaned up here.

Or move the initialization of the list after anything can fail

We could add a simple test to pass in an invalid URI to add a bit more test coverage.

Reply:
Thanks, good catch. Fixed with a test.